cmd/support-callhome-set.go

```diff
@@ -0,0 +1,160 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio-go/v7/pkg/set"
+)
+
+var callhomeFlags = append(globalFlags, cli.BoolFlag{
+	Name:   "dev",
+	Usage:  "Development mode - talks to local SUBNET",
+	Hidden: true,
+})
+
+var callhomeSetCmd = cli.Command{
+	Name:         "set",
+	Usage:        "Enable/disable a callhome option",
+	OnUsageError: onUsageError,
+	Action:       mainCallhomeSet,
+	Before:       setGlobalsFromContext,
+	Flags:        callhomeFlags,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} TARGET option=on|off
+
+OPTIONS:
+  logs - Push MinIO log entries to SUBNET in realtime
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Enable logs callhome for cluster with alias 'play'
+     {{.Prompt}} {{.HelpName}} play logs=on
+
+  2. Disable logs callhome for cluster with alias 'play'
+     {{.Prompt}} {{.HelpName}} play logs=off
+`,
+}
+
+func supportedOptions() set.StringSet {
+	return set.CreateStringSet("logs")
+}
+
+func validOptionValues() set.StringSet {
+	return set.CreateStringSet("on", "off")
+}
+
+// checkCallhomeSyntax - validate arguments passed by a user
+func checkCallhomeSetSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) < 2 {
+		cli.ShowCommandHelpAndExit(ctx, "set", 1) // last argument is exit code
+	}
+}
+
+func validateArgs(args []string) error {
+	for _, arg := range args {
+		_, _, e := getOptionKV(arg)
+		if e != nil {
+			return e
+		}
+	}
+	return nil
+}
+
+func mainCallhomeSet(ctx *cli.Context) error {
+	checkCallhomeSetSyntax(ctx)
+
+	aliasedURL := ctx.Args().Get(0)
+	args := ctx.Args().Tail()
+	fatalIf(probe.NewError(validateArgs(args)), fmt.Sprintf("Invalid arguments: %s", strings.Join(args, ",")))
+
+	for _, option := range ctx.Args().Tail() {
+		// options have already been validated, so third (error) return value can be ignored
+		k, v, _ := getOptionKV(option)
+		if k == "logs" {
+			configureSubnetWebhook(aliasedURL, v == "on")
+		}
+	}
+
+	return nil
+}
+
+func configureSubnetWebhook(aliasedURL string, enable bool) {
+	// Get the alias parameter from cli
+	alias, _ := url2Alias(aliasedURL)
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Unable to initialize admin connection.")
+
+	apiKey := getSubnetAPIKeyFromConfig(alias)
+	if len(apiKey) == 0 {
+		err := fmt.Errorf("Please register the cluster first by running 'mc admin subnet register %s'", alias)
```

Comment:
```suggestion
		e := fmt.Errorf("Please register the cluster first by running 'mc admin subnet register %s'", alias)
```

Go error is `e` and probe.Error is `err`